pkg/services/dashboards/dashboard_service.go

```diff
@@ -93,6 +93,16 @@ func (dr *dashboardServiceImpl) buildSaveDashboardCommand(dto *SaveDashboardDTO,
 		}
 	}
 
+	validateBeforeSaveCmd := models.ValidateDashboardBeforeSaveCommand{
+		OrgId:     dto.OrgId,
+		Dashboard: dash,
+		Overwrite: dto.Overwrite,
+	}
+
+	if err := bus.Dispatch(&validateBeforeSaveCmd); err != nil {
```

Comment:
I can see one case where we may have a problem with doing the validate before save. If validate before save returns a `ErrDashboardVersionMismatch` or `ErrDashboardWithSameNameInFolderExists` the result would be a overwrite warning dialog in UI. If user clicks on overwrite it will popup "you cannot save provisioned dashboard" dialog if the dashboard is provisioned. 

Unsure if this is acceptable or if we need to add some special case handling here? Guess it depends on what use case we're trying to solve.

Reply:
This could only happen when users try to save over a provisioned dashboard. Right?

If that's the case we should give the user `You cannot save provisioned dashboard` dialog.

Reply:
> This could only happen when users try to save over a provisioned dashboard. Right?

Yes

> If that's the case we should give the user You cannot save provisioned dashboard dialog.

Then we need to add a special case that we should validate provisioned dashboard if `err = ErrDashboardVersionMismatch` or `err = ErrDashboardWithSameNameInFolderExists`. Will add some complexity, but absolutely doable. What do you think?

Reply:
So.... :) 

I think we might want to check twice if the dashboard is provisioned? before validateSave and after? since the id might update during validateSave.

Reply:
That will not solve the problem with `ErrDashboardVersionMismatch` or `ErrDashboardWithSameNameInFolderExists`. At the same time it will not allow you to overwrite another dashboard that's not provisioned by updating title of a provisioned dashboard.